benchmarks/high_volume_reads_test.go

```diff
@@ -17,126 +18,100 @@ import (
-
-					if benchCfg.Metrics.EnableCadvisorMetrics {
-						err = metricsClient.Measure(e, metrics.ContainerMemoryWorkingSetBytes(job, samplingRange, annotation))
-						Expect(err).Should(Succeed(), fmt.Sprintf("Failed - %v", err))
-					}
-
-					err = metricsClient.Measure(e, metrics.RequestReadsQPS(job, samplingRange, annotation))
-					Expect(err).Should(Succeed(), fmt.Sprintf("Failed - %v", err))
-					err = metricsClient.Measure(e, metrics.RequestDurationOkQueryRangeAvg(job, samplingRange, annotation))
-					Expect(err).Should(Succeed(), fmt.Sprintf("Failed - %v", err))
-					err = metricsClient.Measure(e, metrics.RequestDurationOkQueryRangePercentile(99, job, samplingRange, annotation))
-					Expect(err).Should(Succeed(), fmt.Sprintf("Failed - %v", err))
-					err = metricsClient.Measure(e, metrics.RequestDurationOkQueryRangePercentile(50, job, samplingRange, annotation))
-					Expect(err).Should(Succeed(), fmt.Sprintf("Failed - %v", err))
-					err = metricsClient.Measure(e, metrics.RequestQueryRangeThroughput(job, samplingRange, annotation))
-					Expect(err).Should(Succeed(), fmt.Sprintf("Failed - %v", err))
-
-					// Ingesters
-					job = benchCfg.Metrics.Jobs.Ingester
-					annotation = metrics.IngesterAnnotation
-
-					err = metricsClient.Measure(e, metrics.ContainerCPU(job, samplingRange, annotation))
-					Expect(err).Should(Succeed(), fmt.Sprintf("Failed - %v", err))
-
-					if benchCfg.Metrics.EnableCadvisorMetrics {
-						err = metricsClient.Measure(e, metrics.ContainerMemoryWorkingSetBytes(job, samplingRange, annotation))
-						Expect(err).Should(Succeed(), fmt.Sprintf("Failed - %v", err))
-					}
-
-					err = metricsClient.Measure(e, metrics.RequestReadsGrpcQPS(job, samplingRange, annotation))
-					Expect(err).Should(Succeed(), fmt.Sprintf("Failed - %v", err))
-					err = metricsClient.Measure(e, metrics.RequestDurationOkGrpcQuerySampleAvg(job, samplingRange, annotation))
-					Expect(err).Should(Succeed(), fmt.Sprintf("Failed - %v", err))
-					err = metricsClient.Measure(e, metrics.RequestDurationOkGrpcQuerySamplePercentile(99, job, samplingRange, annotation))
-					Expect(err).Should(Succeed(), fmt.Sprintf("Failed - %v", err))
-					err = metricsClient.Measure(e, metrics.RequestDurationOkGrpcQuerySamplePercentile(50, job, samplingRange, annotation))
-					Expect(err).Should(Succeed(), fmt.Sprintf("Failed - %v", err))
-
-					err = metricsClient.Measure(e, metrics.RequestBoltDBShipperReadsQPS(job, samplingRange))
-					Expect(err).Should(Succeed(), fmt.Sprintf("Failed - %v", err))
-				}, samplingCfg)
-			})
+		It("should measure metrics", func() {
+			samplingCfg, samplingRange = highVolumeReadsTest.SamplingConfiguration()
+
+			// Sleeping for the first interval so that the data is accurate for the new workload.
+			time.Sleep(samplingCfg.MinSamplingInterval)
+
+			e := gmeasure.NewExperiment(highVolumeReadsTest.Description)
+			AddReportEntry(e.Name, e)
+
+			e.Sample(func(idx int) {
```

Comment:
The sample is skipping collecting metrics from the index gateway component. I remember the time we didn't have to collect these metrics because the component did not exist. However the component is important for the query path and in fact crucial on memory  usage (e.g. with huge boltdb indices being memory mapped)